contracts/smart-account/test/TestSecp256r1.sol

```diff
@@ -0,0 +1,386 @@
+        // u1Points[1] = JPoint(GX, GY, 1); // u1
+        // u1Points[2] = jPointDouble(u1Points[1]);
+        // u1Points[3] = jPointAdd(u1Points[1], u1Points[2]);
+        // avoiding this intermediate step by using it in a single array below
+        // these are pre computed points for u1
+
+        // JPoint[16] memory points;
+        points[0] = JPoint(1, 1, 0);
+        points[1] = JPoint(passKey.pubKeyX, passKey.pubKeyY, 1); // u2
+        points[2] = jPointDouble(points[1]);
+        points[3] = jPointAdd(points[1], points[2]);
+
+        points[4] = JPoint(GX, GY, 1); // u1Points[1]
+        points[5] = jPointAdd(points[4], points[1]);
+        points[6] = jPointAdd(points[4], points[2]);
+        points[7] = jPointAdd(points[4], points[3]);
+
+        points[8] = jPointDouble(points[4]); // u1Points[2]
+        points[9] = jPointAdd(points[8], points[1]);
+        points[10] = jPointAdd(points[8], points[2]);
+        points[11] = jPointAdd(points[8], points[3]);
+
+        points[12] = jPointAdd(points[4], points[8]); // u1Points[3]
+        points[13] = jPointAdd(points[12], points[1]);
+        points[14] = jPointAdd(points[12], points[2]);
+        points[15] = jPointAdd(points[12], points[3]);
+    }
+
+    function jPointAdd(
+        JPoint memory p1,
+        JPoint memory p2
+    ) public pure returns (JPoint memory) {
+        uint256 x;
+        uint256 y;
+        uint256 z;
+        (x, y, z) = jAdd(p1.x, p1.y, p1.z, p2.x, p2.y, p2.z);
+        return JPoint(x, y, z);
+    }
+
+    function jPointDouble(JPoint memory p) public pure returns (JPoint memory) {
+        uint256 x;
+        uint256 y;
+        uint256 z;
+        (x, y, z) = modifiedJacobianDouble(p.x, p.y, p.z);
+        return JPoint(x, y, z);
+    }
+
+    /*
+     * jAdd
+     * @description performs double Jacobian as defined below:
+     * https://hyperelliptic.org/EFD/g1p/auto-code/shortw/jacobian-3/doubling/mdbl-2007-bl.op3
+     */
+    function jAdd(
+        uint256 p1,
+        uint256 p2,
+        uint256 p3,
+        uint256 q1,
+        uint256 q2,
+        uint256 q3
+    ) public pure returns (uint256 r1, uint256 r2, uint256 r3) {
+        if (p3 == 0) {
+            return (q1, q2, q3);
+        } else if (q3 == 0) {
+            return (p1, p2, p3);
+        }
+
+        uint256 u1;
+        uint256 u2;
+        uint256 s1;
+        uint256 s2;
+        assembly {
+            let
+                pd
+            := 0xFFFFFFFF00000001000000000000000000000000FFFFFFFFFFFFFFFFFFFFFFFF
+            let z1z1 := mulmod(p3, p3, pd) // Z1Z1 = Z1^2
+            let z2z2 := mulmod(q3, q3, pd) // Z2Z2 = Z2^2
+
+            u1 := mulmod(p1, z2z2, pd) // U1 = X1*Z2Z2
+            u2 := mulmod(q1, z1z1, pd) // U2 = X2*Z1Z1
+
+            s1 := mulmod(p2, mulmod(z2z2, q3, pd), pd) // S1 = Y1*Z2*Z2Z2
+            s2 := mulmod(q2, mulmod(z1z1, p3, pd), pd) // S2 = Y2*Z1*Z1Z1
+
+            let p3q3 := addmod(p3, q3, pd)
```

Comment:
moving this to line 334 seems to fix the issue forge is having compiling the contracts